### PR TITLE
feat: Update export command to use `--template-dir` and `--template-param`

### PR DESCRIPTION
**Pull request by gadenbuie**
Pairs with https://github.com/posit-dev/shinylive/pull/144
Helps with https://github.com/posit-dev/shinylive/issues/110

This PR adds `template_dir` and `template_params` to `shinylive.export()` and as command line arguments in `shinylive export`.

* `template_dir` or `--template-dir` allows users to customize the `index.html` file used by the exported app. This argument takes a path that, effectively, supercedes the path in the shinylive assets.
* `template_params` or `--template-params` takes a dict of template parameters (in Python) or a JSON string or path to JSON in the CLI. https://github.com/posit-dev/shinylive/pull/144/files introduces a new set of template params, e.g. `{{ title }}`. This argument also covers parameters introduced by custom templates, or future parameters added to the default shinylive template.

We now also use [mustache](https://mustache.github.io/) for templating via chevron to render the templates. In addition to being a more powerful templating engine, this also allows us to use the same template logic for both R and Python. [chevron](https://pypi.org/project/chevron/) appears to be under-maintained but stable and recommended from [the mustache frontpage](https://mustache.github.io/).

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -41,6 +41,7 @@ install_requires =
     lzstring>=1.0.4
     typing-extensions>=4.0.1
     setuptools; python_version>='3.12'
+    pystache>=0.6.5
```
Comment (gadenbuie) [inline setup.cfg:44]: Switch to [chevron](https://github.com/noahmorrison/chevron)

File: shinylive/_utils.py [hunk 2]
```diff
@@ -59,18 +61,16 @@ def listdir_recursive(dir: str | Path) -> list[str]:
...
 
 
-FromTo = Tuple[str, str]
-
-
 def copy_file_and_substitute(
-    src: str | Path, dest: str | Path, *, replacements: Sequence[FromTo]
+    src: str | Path,
+    dest: str | Path,
+    **kwargs: object,
```
Comment (schloerke) [inline shinylive/_utils.py:67]: We should try to eliminate collisions when possible. 

```suggestion
    *,
    template_params: dict[str, object],
```

File: shinylive/_utils.py [hunk 3]
```diff
@@ -59,18 +61,16 @@ def listdir_recursive(dir: str | Path) -> list[str]:
...
 ) -> None:
     with open(src, "r", encoding="utf-8") as fin:
         in_content = fin.read()
-        for from_str, to_str in replacements:
-            in_content = in_content.replace(from_str, to_str)
+        out_content = chevron.render(in_content, data=kwargs)
```
Comment (schloerke) [inline shinylive/_utils.py:71]:
```suggestion
        out_content = chevron.render(in_content, data=template_params)
```

File: shinylive/_app_json.py [hunk 4]
```diff
@@ -108,26 +114,31 @@ def write_app_json(app_info: AppInfo, destdir: Path, html_source_dir: Path) -> N
...
+
+    for template_file in template_files:
+        dest_file = app_destdir / template_file.relative_to(html_source_dir)
+        dest_file.parent.mkdir(parents=True, exist_ok=True)
+
+        if template_file.suffix == ".html":
+            _utils.copy_file_and_substitute(
+                src=template_file,
+                dest=dest_file,
+                **replacements,
```
Comment (schloerke) [inline shinylive/_app_json.py:138]:
```suggestion
                data=replacements,
```

**Review by schloerke — APPROVED** [review]

**Pull request merged**